src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -53,36 +80,31 @@
-     (let [route-config (core/get-master-route-config ::master-service config)
-           path (core/get-master-mount ::master-service route-config)
-           ring-app (core/construct-root-routes puppet-version
-                                                use-legacy-auth-conf
-                                                jruby-service
-                                                get-code-content
-                                                handle-request
-                                                (get-auth-handler)
-                                                environment-class-cache-enabled)
-           routes (comidi/context path ring-app)
-           comidi-handler (comidi/routes->handler routes)
-           ring-handler (comidi/wrap-with-route-metadata comidi-handler routes)]
-       ;; if the webrouting config uses the old-style config where
-       ;; there is a single key with a route-id, we need to deal with that
-       ;; for backward compat.  We have a hard-coded assumption that this route-id
-       ;; must be `master-routes`.  In Puppet Server 2.0, we also supported a
-       ;; key called `invalid-in-puppet-4` in the same route config, even though
-       ;; that key is no longer used for Puppet Server 2.1 and later.  We
-       ;; should be able to remove this hack as soon as we are able to get rid
-       ;; of the legacy routes.
-       (if (and (map? route-config)
-                (contains? route-config :master-routes))
-         (add-ring-handler this
-                           ring-handler
-                           {:route-id :master-routes
-                            :normalize-request-uri true})
-         (add-ring-handler this
-                           ring-handler
-                           {:normalize-request-uri true}))))
-   context)
+
+     ;; if the webrouting config uses the old-style config where
+     ;; there is a single key with a route-id, we need to deal with that
+     ;; for backward compat.  We have a hard-coded assumption that this route-id
+     ;; must be `master-routes`.  In Puppet Server 2.0, we also supported a
+     ;; key called `invalid-in-puppet-4` in the same route config, even though
+     ;; that key is no longer used for Puppet Server 2.1 and later.  We
+     ;; should be able to remove this hack as soon as we are able to get rid
+     ;; of the legacy routes.
+     (if (and (map? route-config)
+              (contains? route-config :master-routes))
+       (add-ring-handler this
+                         ring-handler
+                         {:route-id :master-routes
+                          :normalize-request-uri true})
+       (add-ring-handler this
+                         ring-handler
+                         {:normalize-request-uri true}))
+
+     (register-status
+      "master"
+      (status-core/get-artifact-version "puppetlabs" "puppetserver")
+      1
+      (partial core/v1-status metrics))
```

Comment:
Hmm, good point about `initialize-registry-settings!`.  I hadn't really thought that through.  It does feel a little weird to be updating info for PE-specific metrics, like the file-sync ones, from OSS Puppet Server.  I wonder if it might be better to come up with a way to distribute the list of registry settings across the services that manage them.  Like, maybe we could introduce an `add-to-or-initialize-registry-settings` function on the tk-metrics protocol which could be called multiple times, concatenating the stored list as it goes, before the registry is locked.  If we did that, we could let OSS puppetserver manage the Puppet profiler and http related defaults and maybe move the file-sync related ones over to pe-file-sync if that makes sense.  Maybe we could pick this conversation up further as we do [SERVER-1721](https://tickets.puppetlabs.com/browse/SERVER-1721)?